glass/jax.py

```diff
@@ -0,0 +1,117 @@
+"""Wrapper for JAX RNG with a NumPy-like interface."""
+
+from __future__ import annotations
+
+import math
+from threading import Lock
+from typing import TYPE_CHECKING, TypeAlias
+
+from typing_extensions import Self
+
+import jax.dtypes
+import jax.numpy as jnp
+import jax.random
+from jax.typing import ArrayLike
+
+if TYPE_CHECKING:
+    from jaxtyping import Array, Integer, PRNGKeyArray, Shaped
+
+    RealArray: TypeAlias = Array
+    Size: TypeAlias = int | tuple[int, ...] | None
+
+
+def _size(size: Size, *bcast: Array) -> tuple[int, ...]:
+    """
+    Return a size, which can be a single int or None, as a shape, which
+    is a tuple of int.
+    """
+    if size is None:
+        if bcast:
+            return jnp.broadcast_shapes(*map(jnp.shape, bcast))  # type: ignore[no-any-return]
+        return ()
+    if isinstance(size, int):
+        return (size,)
+    return size
+
+
+class JAXGenerator:
```

Comment:
Looking at this again, the class can now be renamed to `Generator` -- `glass.jax.Generator` looks quite clean.

Reply:
I think this sounds better